lib/src/Base/Stat/HMatrixParameters.cxx

```diff
@@ -126,10 +127,21 @@ String HMatrixParameters::__repr__() const
       << ", recompression epsilon=" << recompressionEpsilon_
       << ", admissibility factor=" << admissibilityFactor_
       << ", clustering algorithm=" << clusteringAlgorithm_
-      << ", compression method=" << compressionMethod_;
+      << ", compression method=" << compressionMethod_
+      << ", factorization method=" << factorizationMethod_;
   return oss;
 }
 
+void HMatrixParameters::setFactorizationMethod(const String &factorization)
+{
+  factorizationMethod_ = factorization;
```

Comment:
Shouldn't this method test whether the string is valid?

Reply:
Check is done in `HMatrixImplementation`

Reply:
Even in the check is done later, shouldn't we catch it as soon as possible rather than let it blow up in the user's face later? I think the test should be moved here (or duplicated if we also need it in HMatrixImplementation).

Reply:
No. Only **advanced users** will use the method. 
If `check` is mandatory then similarly all methods should have the same mechanism + checks for resource map `keys` ...
So there is only *one check*